### ISSUE TITLE
About AU-Pro metrics

### ISSUE THREAD

**Issue by Jay-zzcoder** [caoyunkang/CDO#9]
Thank you for your excellent work! But I meet some problems when I run this code:
Why the metrics AU-Pro is always 0.0 when training like this:
2024-01-30 12:55:23.070 | INFO     | __main__:main:135 - bagel=======i_roc: 96.44
2024-01-30 12:55:23.071 | INFO     | __main__:main:135 - bagel=======p_roc: 99.23
2024-01-30 12:55:23.071 | INFO     | __main__:main:135 - bagel=======p_pro: 0.00
2024-01-30 12:55:23.072 | INFO     | __main__:main:135 - bagel=======threshold: 0.75

**Comment by Jay-zzcoder** [comment]
> hi, the calculation process of AU-PRO is really time-consuming, so I discarded AU-PRO in the training process. Please carefully check the flag `cal_pro` for further information.

Whether the flag 'cal_pro' is set as True or Fasle,  AU_Pro is always0.0
2024-01-31 01:21:15.018 | INFO     | __main__:main:74 - ==========running parameters=============
2024-01-31 01:21:15.019 | INFO     | __main__:main:76 - dataset: mvtec3d
2024-01-31 01:21:15.019 | INFO     | __main__:main:76 - class_name: bagel
2024-01-31 01:21:15.019 | INFO     | __main__:main:76 - img_resize: 256
2024-01-31 01:21:15.019 | INFO     | __main__:main:76 - img_cropsize: 256
2024-01-31 01:21:15.019 | INFO     | __main__:main:76 - num_epochs: 2
2024-01-31 01:21:15.019 | INFO     | __main__:main:76 - validation_epoch: 5
2024-01-31 01:21:15.019 | INFO     | __main__:main:76 - lr: 0.0004
2024-01-31 01:21:15.019 | INFO     | __main__:main:76 - batch_size: 16
2024-01-31 01:21:15.019 | INFO     | __main__:main:76 - vis: True
2024-01-31 01:21:15.019 | INFO     | __main__:main:76 - root_dir: ./result
2024-01-31 01:21:15.019 | INFO     | __main__:main:76 - load_memory: True
2024-01-31 01:21:15.019 | INFO     | __main__:main:76 - cal_pro: True
2024-01-31 01:21:15.019 | INFO     | __main__:main:76 - seed: 111
2024-01-31 01:21:15.019 | INFO     | __main__:main:76 - gpu_id: 0
2024-01-31 01:21:15.019 | INFO     | __main__:main:76 - pure_test: False
2024-01-31 01:21:15.019 | INFO     | __main__:main:76 - backbone: hrnet32
2024-01-31 01:21:15.020 | INFO     | __main__:main:76 - MOM: True
2024-01-31 01:21:15.020 | INFO     | __main__:main:76 - OOM: True
2024-01-31 01:21:15.020 | INFO     | __main__:main:76 - gamma: 2.0
2024-01-31 01:21:15.020 | INFO     | __main__:main:77 - =========================================
2024-01-31 01:21:15.026 | INFO     | utils.training_utils:get_dir_from_args:69 - ===> Root dir for this experiment: ./result/hrnet32-mvtec3d-wi-OOM-wi-MOM/logger/bagel
2024-01-31 01:21:17.217 | INFO     | datasets:get_dataloader_from_args:33 - ===> datasets: mvtec3d, class name/len: bagel/110, batch size: 16
2024-01-31 01:21:23.810 | INFO     | datasets:get_dataloader_from_args:33 - ===> datasets: mvtec3d, class name/len: bagel/244, batch size: 16
CDO:bagel:   0%|                                                                                                                                                                        | 0/2 [00:00<?, ?it/s]2024-01-31 01:21:37.665 | INFO     | __main__:main:131 -

2024-01-31 01:21:37.667 | INFO     | __main__:main:135 - bagel=======i_roc: 57.28
2024-01-31 01:21:37.668 | INFO     | __main__:main:135 - bagel=======p_roc: 94.22
2024-01-31 01:21:37.668 | INFO     | __main__:main:135 - bagel=======p_pro: 0.00
2024-01-31 01:21:37.669 | INFO     | __main__:main:135 - bagel=======threshold: 1.59
